### PR TITLE
Rest security

### PR DESCRIPTION
**Pull request by nicolas2lee**
Update method copyFrom in class User by using java reflection 

Update enum Role, two roles only {admin, user} @shine1982

### CODE REVIEW

File: src/main/java/fr/ospiea/oscra/user/object/User.java [hunk 1]
```diff
@@ -56,12 +146,71 @@ public void setPhoneNumber(String phoneNumber) {
...
+                ", supplementaryAddress='" + supplementaryAddress + '\'' +
+                ", postalCode='" + postalCode + '\'' +
+                ", city='" + city + '\'' +
+                ", phoneNumber='" + phoneNumber + '\'' +
+                ", birthday=" + birthday +
+                ", position='" + position + '\'' +
+                ", fixNumber='" + fixNumber + '\'' +
+                ", civility=" + civility +
+                '}';
+    }
```
Comment (shine1982) [inline src/main/java/fr/ospiea/oscra/user/object/User.java]: I don't think this method is useful.

File: src/main/java/fr/ospiea/oscra/restsecurity/ApplicationSecurity.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                .failureHandler(authenticationFailureHandler)
+                .and()
+                .logout().permitAll()
+                .logoutRequestMatcher(new AntPathRequestMatcher("/api/logout","POST"))
+                .logoutSuccessHandler(logoutSuccessHandler)
+                .and()
+                .sessionManagement()
+                .maximumSessions(1);
+        http.authorizeRequests().antMatchers("/api/**").authenticated();
+
```
Comment (shine1982) [inline src/main/java/fr/ospiea/oscra/restsecurity/ApplicationSecurity.java:64]: do we still need to write '/api' after configure it in properties?

**Review by shine1982 — APPROVED** [review]

**Comment by nicolas2lee** [comment]
updated @shine1982

**Comment by shine1982** [comment]
Is this brach should be merged? You can merge it if you want.

**Pull request merged**